jme3-glsl-highlighter/src/com/jme3/gde/glsl/highlighter/lexer/GlslKeywordLibrary.java

```diff
@@ -0,0 +1,450 @@
+        builtinFunctions.add("imageSize");
+        builtinFunctions.add("imageLoad");
+        builtinFunctions.add("imageStore");
+        builtinFunctions.add("imageAtomicAdd");
+        builtinFunctions.add("imageAtomicMin");
+        builtinFunctions.add("imageAtomicMax");
+        builtinFunctions.add("imageAtomicAnd");
+        builtinFunctions.add("imageAtomicOr");
+        builtinFunctions.add("imageAtomicXor");
+        builtinFunctions.add("imageAtomicExchange");
+        builtinFunctions.add("imageAtomicCompSwap");
+        builtinFunctions.add("dFdx");
+        builtinFunctions.add("dFdy");
+        builtinFunctions.add("fwidth");
+        builtinFunctions.add("interpolateAtCentroid");
+        builtinFunctions.add("interpolateAtSample");
+        builtinFunctions.add("interpolateAtOffset");
+        builtinFunctions.add("noise1");
+        builtinFunctions.add("noise2");
+        builtinFunctions.add("noise3");
+        builtinFunctions.add("noise4");
+        builtinFunctions.add("EmitStreamVertex");
+        builtinFunctions.add("EndStreamPrimitive");
+        builtinFunctions.add("EmitVertex");
+        builtinFunctions.add("EndPrimitive");
+        builtinFunctions.add("barrier");
+        builtinFunctions.add("memoryBarrier");
+        builtinFunctions.add("memoryBarrierAtomicCounter");
+        builtinFunctions.add("memoryBarrierBuffer");
+        builtinFunctions.add("memoryBarrierShared");
+        builtinFunctions.add("memoryBarrierImage");
+        builtinFunctions.add("groupMemoryBarrier");
+    }
+    
+    public static KeywordType lookup(String s){
+        KeywordType returnType = null;
+        for (String primitive : basicTypes){
+            if (primitive.startsWith(s)){
+                if (primitive.equals(s)){
+                    returnType = KeywordType.BASIC_TYPE;
+                    break;
+                }else
+                    returnType = KeywordType.UNFINISHED;
+            }
+        }
+        for (String var : builtinVariables){
+            if (var.startsWith(s)){
+                if (var.equals(s)){
+                    returnType = KeywordType.BUILTIN_VARIABLE;
+                    break;
+                }else
+                    returnType = KeywordType.UNFINISHED;
+            }
+        }
+        for (String func : builtinFunctions){
+            if (func.startsWith(s) && (returnType == KeywordType.UNFINISHED || returnType == null)){
+                if (func.equals(s)){
+                    returnType = KeywordType.BUILTIN_FUNCTION;
+                    break;
+                }else
+                    returnType = KeywordType.UNFINISHED;
+            }
+        }
+        for (String keyword : keywords){
+            if (keyword.startsWith(s) && (returnType == KeywordType.UNFINISHED || returnType == null)){
+                if (keyword.equals(s)){
+                    returnType = KeywordType.KEYWORD;
+                    break;
+                }
```

Comment:
Another #155 here:
- Every if/else clause should be wrapped with {}'s because it improves readability (you always see braces) and it prevents errors from growing where someone is adding a line into the if which actually isn't part of the if.
- Run Netbeans Source -> Format on this file (spaces between `){` missing, in this line the else is not on the same line as the `}` etc. Just a bit of consistent style here.